File: echo.js
var kexec = require('kexec');

message = process.argv[2];
process.stdout.write(message.charAt(0));

remainder = message.substr(1, message.length-1);
if (remainder) {
  kexec("python echo.py '"+remainder+"'");
} else {
  console.log();
}

File: echo.pl
$message = @ARGV[0];
print substr($message, 0, 1);

$remainder = substr($message, 1, length($message)-1);
if (!$remainder eq "") {
  $cmd = "ruby echo.rb '" . $remainder . "'";
  exec($cmd);
}
else {
  print "\n";
}
